### PR TITLE
Create all variables directly from connection and independent of dataframes

### LINKED ISSUE #923: Refactor variables
A Variable should have two things: an indices table and JuMP vector.
Currently, many variables are still in the old format (see `src/variables/investments.jl`).
To solve this, we have to:

- Create variables indices in `compute_variables_indices` for all variables
- Directly use `connection` instead of `dataframes` to construct indices
- Create JuMP variables from the indices for all variables (which means updating the whole model)

Expected challenges:

- Some sets are heavily dependent on other sets, so the logic needs to be translated to DuckDB tables
- Changing the `@variable` definition to use the table requires:
  - Changing how some values are computed (e.g., `weight_for_assets_investment_discounts`) to use DuckDB tables as well
  - Changing all places that use the variable to loop over tables

### PR DESCRIPTION
**Pull request by abelsiqueira**
**Blocked by #925**

Create all variables with connection instead of dataframes variable.

With the `lookup` created in #925, this allow us to

Close #923

### CODE REVIEW

File: src/tmp.jl [hunk 1]
```diff
@@ -157,6 +159,148 @@ function tmp_create_partition_tables(connection)
...
+    DuckDB.end_row(appender)
+end
+
+function tmp_create_lowest_resolution_table(connection)
+    # These are generic tables to be used to create some variables and constraints
+
+    # The logic:
+    # - t_union has groups in order, then s:e ordered by s increasing and e decreasing
+    # - in a group (asset, year, rep_period) we can take the first range then
+    # - continue in the sequence selecting the next largest e
```
Comment (abelsiqueira) [inline src/tmp.jl:252]: Remove or update

Comment (abelsiqueira) [reply src/tmp.jl:252]: Use

- 1:4 5:8 9:12
- 1:3 4:4 5:9 10:12

1 2 3 4 
o o o o
o o o
           o o o o
                o o o o
                o o o o o
                              o o o o
                                 o o o

o o o o
               o

**Review by gnawin — APPROVED** [review]
LGTM

**Comment by abelsiqueira** [comment]
Thanks!

**Pull request merged**